### PR TITLE
Add support for server controlled resync on switching sync regions

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/1582/add-sync-region-id-header-to-http-requests

### CODE REVIEW

File: app/src/androidTest/java/org/simple/clinic/user/UserSessionAndroidTest.kt [hunk 1]
```diff
@@ -111,4 +112,31 @@ class UserSessionAndroidTest {
...
+    val newFacility = TestData.facility(
+        uuid = UUID.fromString("56ddc7df-6a81-42bc-8659-78b0ccb51edd"),
+        syncGroup = "bb92c082-d749-478a-bd2a-fe7f10055ef0"
+    )
+    facilityRepository
+        .save(listOf(newFacility))
+        .blockingAwait()
+
+    facilityRepository
+        .setCurrentFacility(newFacility)
```
Comment (Honey14) [inline app/src/androidTest/java/org/simple/clinic/user/UserSessionAndroidTest.kt:128]: `setCurrentFacility` is expecting facilityUuid here. Hence the test is failing.

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good overall. One android test is failing. PTAL

File: app/src/androidTest/java/org/simple/clinic/user/UserSessionAndroidTest.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/androidTest/java/org/simple/clinic/user/UserSessionAndroidTest.kt:128]: Fixing.

**Review by Honey14 — APPROVED** [review]
Looks good

**Pull request merged**